packages/itwin/grouping-mapping-widget/src/widget/components/GroupPropertyAction.tsx

```diff
@@ -68,98 +91,138 @@ export const quantityTypesSelectionOptions: SelectOption<QuantityType>[] = [
   { value: QuantityType.Volume, label: "Volume" },
   { value: QuantityType.Undefined, label: "No Quantity Type" },
 ];
-interface Property {
-  name: string;
-  label: string;
-  type: string;
-}
 
-interface NavigationProperty {
-  navigationName: string;
-  rootClassName: string;
+interface PropertyMetaData {
+  // Display label
+  label: string;
+  // ECProperty source schema:class
+  schema: string;
+  className: string;
+  categoryLabel: string;
+  // Property Type
+  type: DataType;
+  // The actual ECClass name of the property
+  actualECClassName: string;
+  // The parent class of the property
+  parentPropertyClassName: string | undefined;
+  // ECProperty type traversal
+  propertyTraversal: Array<string>;
+  // The type of primitive navigation from Presentation
+  primitiveNavigationClass: string;
+  key: string;
 }
 
-const extractPrimitive = (
-  propertiesField: PropertiesField,
-  classToPropertiesMapping: Map<string, Property[]>,
-  navigation?: NavigationProperty
-) => {
-  // There are rare cases which only happens in multiple selections where it returns more than one.
-  // This also checks if this property comes from a navigation property
-  const className =
-    navigation?.rootClassName ??
-    propertiesField.properties[0].property.classInfo.name;
-  // Sometimes class names are not defined. Type error. Not guaranteed.
-  if (!className) {
-    return;
-  }
-
-  if (!classToPropertiesMapping.has(className)) {
-    classToPropertiesMapping.set(className, []);
+const convertType = (type: string): DataType => {
+  switch (type) {
+    case "int":
+    case "enum":
+    case "long":
+      return DataType.Integer;
+    case "boolean":
+      return DataType.Boolean;
+    case "double":
+      return DataType.Double;
+    case "string":
+      return DataType.String;
+    case "number":
+      return DataType.Number;
+    default:
+      return DataType.Undefined;
```

Comment:
I think it would be better to default to `DataType.String`. Sending `DataType.Undefined` to Reporting API will cause a BadRequest.

Reply:
Done.